File: day01/nim/part1.py
#!/usr/bin/env python3


def read_lines(fname):
    with open(fname) as f:
        return f.read().strip().splitlines()


def main():
    lines = read_lines("input.txt")
    numbers = [int(e) for e in lines]
    result = sum(numbers)

    print(result)

##############################################################################

if __name__ == "__main__":
    main()

File: day01/nim2/part1.nim
#!/usr/bin/env nimbang
#off:nimbang-args c -d:release
#nimbang-settings hideDebugInfo

import std/strutils    # strip, split, join
import std/sequtils    # toSeq(iter), map, filter
import std/math        # sum

proc main() =
  # let fname = "example.txt"
  let fname = "input.txt"

  let result = readFile(fname).strip.splitLines.map(parseInt).sum

  echo result

# ############################################################################

when isMainModule:
  main()
